src/main/java/org/solid/testharness/accesscontrol/AccessDatasetAcp.java

```diff
@@ -60,28 +62,34 @@
 
     public AccessDatasetAcp(final List<AccessRule> accessRules, final String aclUri) {
         final String namespace = aclUri + "#";
+        final IRI accessControlResource = iri(aclUri);
         if (!accessRules.isEmpty()) {
             final ModelBuilder builder = new ModelBuilder();
             // add document namespace and any others needed
             builder.setNamespace(PREFIX, namespace);
             builder.setNamespace(ACP.PREFIX, ACP.NAMESPACE);
+            builder.setNamespace(ACL.PREFIX, ACL.NAMESPACE);
             if (accessRules.stream().anyMatch(rule -> rule.getType() == AccessRule.AgentType.GROUP)) {
                 builder.setNamespace(VCARD.PREFIX, VCARD.NAMESPACE);
             }
+
             final AtomicInteger i = new AtomicInteger();
             accessRules.forEach(rule -> {
                 i.addAndGet(1);
                 final List<String> accessModes = rule.getAccess().stream()
                         .filter(mode -> !controlModes.contains(mode))
                         .collect(Collectors.toList());
                 final List<String> controlAccessModes = extractControlModes(rule);
+                final IRI accessControlNode = iri(namespace, "accessControl" + i);
                 final IRI policyNode = iri(namespace, "policy" + i);
                 final IRI ruleNode = iri(namespace, "rule" + i);
-                builder.subject(iri(aclUri))
+                builder.add(accessControlResource, ACP.accessControl, accessControlNode)
```

Comment:
In the current draft, inheritance happens at the AccessControl level, not the Policy level